### PR TITLE
http: require HTTP/0.9 to be explicitly allowed to get accepted

### LINKED ISSUE #2873: Make HTTP/0.9 opt-in
### I did this

Use curl to get data from a server that isn't responding with a proper HTTP/1 or HTTP/2 response.

curl will gladly hand over the received data as HTTP body. This is because it supports "HTTP/0.9", which is how HTTP used to work before 1.0 (1996): that's just body bytes that ends with a closed connection.

### I expected the following

It should report an error if the response is not HTTP protocol compliant. I suspect this "liberal in what you accept" and treating everything as HTTP/0.9 might be surprising to users and I'm also afraid it could cause some unexpected behaviors.

HTTP/0.9 support should therefor be done opt-in. [mentioned on the mailing list in July 2018](https://curl.haxx.se/mail/lib-2018-07/0006.html) - no objections there.

### curl/libcurl version

curl 7.61.1 still has it supported by default.

### operating system

All

### Releated

HTTP/0.9 detection was improved in #2872 after which fixing this issue should be pretty easy.

### PR DESCRIPTION
**Pull request by bagder**
CURLOPT_HTTP09 and --http0.9 are added for this purpose.

The default is now kept as before, but added option to explicitly allow it and added a mention to DEPRECATE.md to switch the default to not allowed by support 6 months after the 7.64.0 release.

Fixes #2873

### CODE REVIEW

**Comment by bagder** [comment]
Oops, it broke a bunch of tests...

**Comment by bagder** [comment]
The fuzzer hits the 50 minute limit... :-1:

File: docs/DEPRECATE.md [hunk 1]
```diff
@@ -64,3 +64,18 @@ revert if need be.
...
+
+### State
+
+curl 7.64.0 introduces options to disable/enable support for this protocol
+version. The default remains supported for now.
+
+### Removal
+
+The support for HTTP/0.9 will be switched to disabled by default in 6 months,
+in the September 2019 release (possilby called curl 7.68.0).
```
Comment (danielgustafsson) [inline docs/DEPRECATE.md]: s/possilby/possibly/

File: docs/cmdline-opts/http0.9.d [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Added:
+Help: Allow HTTP 0.9 responses
+---
+Tells curl to be fine with HTTP version 0.9 response.
+
+HTTP/0.9 is a completely headerless response and therefore you can also
+connect with this to non-HTTP servers and still get a response since curl will
+simply transparently downgrade - if allowed.
+
+A future curl version will deny continuing if the resposne isn't at least
```
Comment (danielgustafsson) [inline docs/cmdline-opts/http0.9.d]: s/resposne/response/

**Comment by bagder** [comment]
Thanks @danielgustafsson, fixed the spelling errors and I've renamed the libcurl option to `CURLOPT_HTTP09_ALLOWED`. Oops, forgot that there were some test failures left to fix...

**Pull request closed without merging**